### PR TITLE
Doc. fix incorrect arrayFold example

### LINKED ISSUE #55920: [Docs] `arrayFold` incorrect of confusing example
[The documentation for `arrayFold`](https://clickhouse.com/docs/en/sql-reference/functions/array-functions#arrayfold) make it seem that `arrayFold` works like a regular reduce function in programming languages.

There is a usage example:
```sql
SELECT arrayFold( x,acc -> acc + x*2,  [1, 2, 3, 4], toInt64(3)) AS res;
```
With this result:
```
┌─arrayFold(lambda(tuple(x, acc), plus(acc, multiply(x, 2))), [1, 2, 3, 4], toInt64(3))─┐
│                                                                                     3 │
└───────────────────────────────────────────────────────────────────────────────────────┘
```

If `arrayFold` really works like a reduce function, then the result should be 23 (`3 + 1*2 + 2*2 + 3*2 + 4*2`).

Is the example wrong, or am I missing something?

I can't check this function with a real ClickHouse instance, because [the playground](https://play.clickhouse.com/play?user=play#U0VMRUNUIGFycmF5Rm9sZCggeCxhY2MgLT4gYWNjICsgeCoyLCAgWzEsIDIsIDMsIDRdLCB0b0ludDY0KDMpKSBBUyByZXM7) doesn't support the function.

### PR DESCRIPTION
**Pull request by den-crane**
closes: https://github.com/ClickHouse/ClickHouse/issues/55920

### Changelog category (leave one):
- Documentation (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit bd934575ea94a91bd6985bbc13fa4b6ab7984a12 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/55926/bd934575ea94a91bd6985bbc13fa4b6ab7984a12/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6610684469">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/55926/bd934575ea94a91bd6985bbc13fa4b6ab7984a12/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: docs/en/sql-reference/functions/array-functions.md [hunk 1]
```diff
@@ -1144,11 +1144,32 @@ SELECT arrayFold( x,acc -> acc + x*2,  [1, 2, 3, 4], toInt64(3)) AS res;
...
 ``` text
-┌─arrayFold(lambda(tuple(x, acc), plus(acc, multiply(x, 2))), [1, 2, 3, 4], toInt64(3))─┐
-│                                                                                     3 │
-└───────────────────────────────────────────────────────────────────────────────────────┘
+┌─res─┐
+│  23 │
+└─────┘
+```
+
+**Example with the fibonacci sequence**
```
Comment (antaljanosbenjamin) [inline docs/en/sql-reference/functions/array-functions.md:1152]:
```suggestion
**Example with the Fibonacci sequence**
```

Comment (antaljanosbenjamin) [reply docs/en/sql-reference/functions/array-functions.md]: I am not sure if the the aspell dict has to be extended with `Fibonacci` or not.

**Review by antaljanosbenjamin — CHANGES_REQUESTED** [review]
Apart from the above mentioned thing it looks fine.

**Review by antaljanosbenjamin — APPROVED** [review]

**Pull request merged**